lib/src/signed_video_h26x_auth.c

```diff
@@ -187,20 +188,18 @@ prepare_for_link_and_gop_hash_verification(signed_video_t *self, h26x_nalu_list_
 
     hash_to_add = item->need_second_verification ? item->second_hash : item->hash;
 
-    if (list_idx == 0) {
-      openssl_init_hash(self->crypto_handle, true);
+    if (list_idx == -1) {
+      update_fallback_gop_hash(self, hash_to_add);
+    }
+    if (list_idx + hash_size > self->gop_info->hash_list_size) {
+      list_idx = -1;
+      initialize_and_update_gop_level_hash(self, hash_list, list_idx);
     }
     // Copy the hash to the list if there is enough space
-    if (list_idx >= 0 && (list_idx + hash_size <= self->gop_info->hash_list_size) &&
-        item->nalu->is_last_nalu_part) {
-      update_fallback_gop_hash(self, hash_to_add);
-      // memcpy(&hash_list[list_idx], hash_to_add, hash_size);
+    if (list_idx >= 0) {
+      memcpy(&hash_list[list_idx], hash_to_add, hash_size);
```

Comment:
Now you've added back copying hashes to a list/memory. Then you need to check for space so you do not write outside memory.